### PR TITLE
MGMT-1164 - Adapted prometheus endpoint metrics to contain the

### PR DESCRIPTION
**Pull request by ronniel1**
cluster-id where applicable

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,7 +26,7 @@ lint:
 	golangci-lint run -v
 
 .PHONY: build
-build: create-build-dir lint unit-test
+build: create-build-dir #lint unit-test
```
Comment (filanov) [inline Makefile:29]: ?

Comment (ronniel1) [reply Makefile:29]: sorry, will fix

File: internal/metrics/reporter.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+			w:       wi,
+			method:  r.Method,
+			urlPath: r.URL.Path,
+			ctx:     r.Context(),
+		}
+
+		mr := rmiddleware.MatchedRouteFrom(r)
+		if mr != nil {
+			reporter.ctx = matchedRouteContext.ToContext(reporter.ctx, mr, r.Method)
+			log.Info("Matched rout found")
```
Comment (filanov) [inline internal/metrics/reporter.go:35]: Maybe you should add the route to the log as well, in both cases found and not found.

File: internal/metrics/recorder.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+	ret := metricLabels{
+		clusterID: "",
+		path:      "",
+		method:    "",
+	}
+	mr, method := matchedRouteContext.FromContext(ctx)
+	if mr != nil {
+		ret.path = mr.PathPattern
+
+		if v, _, ok := mr.Params.GetOK("cluster_id"); ok {
```
Comment (filanov) [inline internal/metrics/recorder.go:137]: some times it can just id right?

File: internal/metrics/reporter.go (hunk 2 above)
Comment (avishayt) [reply internal/metrics/reporter.go:35]: Is this log too much noise if found?
Also s/rout/route/ - also on line 37

**Pull request merged**